core/app.cc

```diff
@@ -217,12 +217,19 @@ application::application(const std::string& command,
         throw launch_error("Failed to load object: " + command);
     }
 
-    _main = _lib->lookup<int (int, char**)>(main_function_name.c_str());
-    if (!_main) {
-        _entry_point = reinterpret_cast<void(*)()>(_lib->entry_point());
-    }
-    if (!_entry_point && !_main) {
-        throw launch_error("Failed looking up main");
+    if (_lib->is_statically_linked()) {
```

Comment:
Following my observation on liblua.so that something can be both a shared library with main() (this was the classic way to build stuff for OSv...) and have a broken, silly, I wonder if it doesn't make more sense to first check if we can find a main(), and only if we can't, fall back to using the entry point.
In fact, that's what the original code below did - I presume for PIEs - it first tried to find main() and only if it couldn't it set  _entry_point. Why can't or shouldn't we do the same for statically linked executable? Isn't the _entry_point calculation the same?

Reply:
Yeah, I need to think it through how to best address it.

Reply:
Addressed by better names and comments and some fixes in the latest version of this PR.